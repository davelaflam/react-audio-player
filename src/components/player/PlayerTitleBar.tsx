import React from 'react'
import { Card, CardContent, Box, Typography, Avatar, List, ListItem } from '@mui/material'

import { PlayerTitleBarProps } from '../../types'

const PlayerTitleBar: React.FC<PlayerTitleBarProps> = ({
  albumCoverPath,
  albumArtist,
  albumTitle,
  albumYear,
  bandMembers,
}) => {
  return (
    <Card sx={{ width: '100%', mb: 2, borderRadius: 2 }}>
      <CardContent
        sx={{
          display: 'flex',
          flexDirection: { xs: 'column', md: 'row' },
          alignItems: { xs: 'center', md: 'flex-start' },
          justifyContent: 'space-between',
        }}
      >
        <Box
          sx={{
            display: 'flex',
            flexDirection: { xs: 'column', md: 'row' },
            alignItems: 'center',
            width: '100%',
          }}
        >
          <Avatar
            variant="rounded"
            src={albumCoverPath}
            alt="Album Cover"
            sx={{
              width: { xs: 100, md: 150 },
              height: { xs: 100, md: 150 },
              mr: { md: 2 },
              mb: { xs: 2, md: 0 },
            }}
          />
          <Box sx={{ flex: 1, textAlign: { xs: 'center', md: 'left' } }}>
            <Box
              sx={{
                display: 'flex',
                alignItems: 'center',
                justifyContent: { xs: 'center', md: 'flex-start' },
              }}
            >
              <Typography
                variant="h6"
                sx={{
                  fontWeight: 'bold',
                  color: 'primary.main',
                  fontSize: { xs: '13px', md: '1.5rem' },
                }}
              >
                {albumArtist}
              </Typography>
              {albumTitle && (
                <>
                  <Typography
                    variant="h6"
                    sx={{
                      mx: 1,
                      color: 'grey.500',
                      fontSize: { xs: '13px', md: '1.25rem' },
                    }}
                  >
                    -
                  </Typography>
                  <Typography variant="h6" sx={{ fontSize: { xs: '13px', md: '1.25rem' } }}>
                    {albumTitle}
                  </Typography>
                </>
              )}
            </Box>
            <List
              dense
              sx={{
                pt: 1,
                display: 'flex',
                flexDirection: { xs: 'row', md: 'column' },
                flexWrap: { xs: 'wrap', md: 'nowrap' },
                gap: { xs: 1, md: 0.5 },
                justifyContent: { xs: 'center', md: 'flex-start' },
                p: 0,
              }}
            >
              {bandMembers.map((member, idx) => (
                <ListItem key={idx} sx={{ width: 'auto', p: 0 }}>
                  <Typography variant="body2" sx={{ fontSize: { xs: '12px', md: '14px' } }}>
                    {member}
                  </Typography>
                </ListItem>
              ))}
            </List>
          </Box>
        </Box>

        <Box sx={{ display: { xs: 'none', md: 'block' }, alignSelf: 'center' }}>
          <Typography variant="subtitle1" sx={{ color: 'grey.500', whiteSpace: 'nowrap', fontSize: { md: '1rem' } }}>
            {albumYear}
          </Typography>
        </Box>
      </CardContent>

      <Box sx={{ display: { xs: 'block', md: 'none' }, textAlign: 'center', mt: 1 }}>
        <Typography variant="subtitle2" sx={{ color: 'grey.500', fontSize: '12px' }}>
          {albumYear}
        </Typography>
      </Box>
    </Card>
  )
}

export default PlayerTitleBar
