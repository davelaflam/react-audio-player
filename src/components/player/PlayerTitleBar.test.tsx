import { render } from '@testing-library/react'
import PlayerTitleBar from './PlayerTitleBar'

test('PlayerTitleBar matches snapshot', () => {
  const { container } = render(
    <PlayerTitleBar
      albumCoverPath="dummy-cover.jpg"
      albumArtist="Dummy Artist"
      albumTitle="Dummy Album"
      albumYear="2023"
      bandMembers={[
        'Stephan Roy - Vocals',
        'Brian Gunter - Guitar, Cello',
        'Dave LaFlam - Electric/Upright Bass',
        'James Beasley - Drums',
      ]}
    />
  )
  expect(container).toMatchSnapshot()
})
