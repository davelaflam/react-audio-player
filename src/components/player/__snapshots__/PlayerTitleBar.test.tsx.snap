// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerTitleBar matches snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-xds4ak-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiCardContent-root css-11x8u7i-MuiCardContent-root"
    >
      <div
        class="MuiBox-root css-19axrzo"
      >
        <div
          class="MuiAvatar-root MuiAvatar-rounded css-1jetft3-MuiAvatar-root"
        >
          <img
            alt="Album Cover"
            class="MuiAvatar-img css-1su80sj-MuiAvatar-img"
            src="dummy-cover.jpg"
          />
        </div>
        <div
          class="MuiBox-root css-1n03rcd"
        >
          <div
            class="MuiBox-root css-1fx0y3e"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-17opwfl-MuiTypography-root"
            >
              Dummy Artist
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-h6 css-wpapln-MuiTypography-root"
            >
              -
            </h6>
            <h6
              class="MuiTypography-root MuiTypography-h6 css-hp6zy8-MuiTypography-root"
            >
              Dummy Album
            </h6>
          </div>
          <ul
            class="MuiList-root MuiList-padding MuiList-dense css-1a1tr8s-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-14z2y7c-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-nvb5dt-MuiTypography-root"
              >
                Stephan Roy - Vocals
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-14z2y7c-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-nvb5dt-MuiTypography-root"
              >
                Brian Gunter - Guitar, Cello
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-14z2y7c-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-nvb5dt-MuiTypography-root"
              >
                Dave LaFlam - Electric/Upright Bass
              </p>
            </li>
            <li
              class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-14z2y7c-MuiListItem-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-nvb5dt-MuiTypography-root"
              >
                James Beasley - Drums
              </p>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="MuiBox-root css-1cf4li4"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-lg29tt-MuiTypography-root"
        >
          2023
        </h6>
      </div>
    </div>
    <div
      class="MuiBox-root css-1cfhbp2"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle2 css-1sibp0x-MuiTypography-root"
      >
        2023
      </h6>
    </div>
  </div>
</div>
`;
